cmd/ingest/ingest.go

```diff
@@ -61,6 +59,23 @@ func main() {
 		log.Fatal(err)
 	}
 
+	key, err := config.GetC4GHKey()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Putting the comment here as the code is spread out a bit.

The reason we didn't do it this way initially was to mitigate one of the flaws in the python version, that the key is always unlocked when the app is running.

That's why I suggested moving the unlocking of the key to the `config` part, since it is used by more than one component, while still returning the keypath and the passphrase to the app. That way we test that the key can be unlocked and we fail early if it can't. But we only unlock it when it is time to use it.

Reply:
I don't have a strong opinion on keeping a single key fetch, but I don't see that helping security much (for most attacks, it's likely that the attacker has access to the file and can more easily grab the passphrase from the environment than analyze a memory dump to try to locate something that looks like a key).

If we move it back inside the loop, we might as well clear the array used for the key when we're done with it.